models/company_gateway/models/llm/llm.py

```
@@ -479,6 +484,72 @@ def _extract_response_tool_calls(
+        except Exception as e:
+            # If request fails and we have response_format, try without it
+            if data and "response_format" in data and ("400" in str(e) or "Bad Request" in str(e)):
+                logger.warning(f"Request failed with response_format, retrying without structured output: {e}")
+                fallback_data = data.copy()
+                response_format = fallback_data.pop("response_format", None)
+                logger.info(f"Removed response_format '{response_format}' from request, falling back to text output")
+                
+                try:
+                    response = self._make_request(method, url, headers, fallback_data, stream, timeout)
+                    logger.info("Fallback request succeeded without structured output")
+                    return response
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)